pkg/postgresql/postgresql.go

```diff
@@ -360,15 +364,15 @@ func (p *Manager) SetupRoles() error {
 	ctx, cancel := context.WithTimeout(context.Background(), p.requestTimeout)
 	defer cancel()
 
-	if p.suUsername == p.replUsername {
+	if p.suUsername == p.replUsername && p.suAuthMethod != "trust" && p.replAuthMethod != "trust" {
```

Comment:
We should add the replication role to superuser also when authmethod is trust. What shouldn't be done is to set a password. Probably alterRole method should be improved/changed to work with empty passwords.

The same also for the call to `createRole` below (in `creating replication role`).